README.md

```diff
@@ -19,40 +18,38 @@ jobs:
     name: Rust project
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@master
-      - name: Install nightly
+      - name: Checkout repository
+        uses: actions/checkout@master
+
+      - name: Install stable toolchain
         uses: actions-rs/toolchain@v1
         with:
-            toolchain: nightly
-            override: true
+          toolchain: stable
+          override: true
 
-      # `cargo check` command here will use installed `nightly`
-      # as it set as an "override" for current directory
+      - name: Run cargo-tarpaulin
+        uses: actions-rs/tarpaulin@v1
+        with:
+          version: '0.9.0'
+          tapaulin-args: '-- --test-threads 1'
+
+      - name: Upload to codecov.io
+        uses: codecov/codecov-action@v1.0.2
+        with:
+          token: ${{secrets.CODECOV_TOKEN}}
 
-      - name: Run cargo check
-        uses: actions-rs/cargo@v1
+      - name: Archive code coverage results
+        uses: actions/upload-artifact@v1
         with:
-          command: check
+          name: code-coverage-report
+          path: cobertura.xml
 ```
 
 See [additional recipes here](https://github.com/actions-rs/meta).
 
 ## Inputs
 
-* `toolchain` (*required*): Toolchain name, see [rustup page](https://github.com/rust-lang/rustup.rs#toolchain-specification) for details.\
-  Examples: `stable`, `nightly`, `nightly-2019-04-20`
-* `target`: Additionally install specific target for this toolchain (ex. `x86_64-apple-darwin`)
-* `default`: Set installed toolchain as default (executes `rustup toolchain default {toolchain}`)
-* `override`: Set installed toolchain as an override for current directory
-
-## Components
-
-If you are going to install `clippy`, `rustfmt` or any other [rustup component](https://rust-lang.github.io/rustup-components-history/),
-it might not be available in latest `nightly` build;
-check out the [`actions-rs/components-nightly`](https://github.com/actions-rs/components-nightly) Action,
-which makes this process much easier.
-
-## Notes
-
-As `rustup` is not installed by default for [macOS environments](https://help.github.com/en/articles/virtual-environments-for-github-actions)
-at the moment (2019-09-13), this Action will try its best to install it before any other operations.
+* `version`: The version of `cargo-tarpaulin` that will be installed.
+* `run-types`: The type of tests to run (`Tests`, or `Doctests`). Runs all by default. May be overridden by `args`.
+* `timeout`: The timeout, in seconds, before cancelling execution of a long running test. May be overriden by `args`.
```

Comment:
I'm not 100% sure about this, but I'm going to have a think about it. I think there's some cases where a "user friendly" option could be better, but then maybe the best place for that is in the upstream package.

Reply:
Coming back to this, I think allowing an `args` input and special casing some of the more common inputs makes sense (to make them more visible where a program has a lot of options). Otherwise, we'd need to keep adding new inputs to the config file as the programs grow more features and options.